mfc9332cdw: init at 1.1.3-0/1.1.4-0

felixsinger opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

felixsinger committed:
mfc9332cdw: init at 1.1.3-0/1.1.4-0

felixsinger committed:
maintainers: Add felixsinger

NickHu reviewed (changes requested):
Please reword the commit message to "maintainers: add felixsinger"

("myself" is a big ambiguous when looking at the git commit log for nixpkgs!)

NickHu reviewed (changes requested):
Please squash into previous commit

NickHu commented:
@felixsinger Thanks a lot for your contribution! I've left a couple of comments

mweinelt commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+      make -C brcupsconfig/ clean
+    '';
+
+    meta = {
```
```suggestion
    meta = with lib; {
```

mweinelt commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+    meta = {
+      description = "Brother ${lib.strings.toUpper model} CUPS wrapper driver";
+      homepage = "http://www.brother.com";
+      license = stdenv.lib.licenses.gpl2;
```
```suggestion
      license = licenses.gpl2;
```

mweinelt commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+      homepage = "http://www.brother.com";
+      license = stdenv.lib.licenses.gpl2;
+      platforms = [ "x86_64-linux" "i686-linux" ];
+      maintainers = [ stdenv.lib.maintainers.felixsinger ];
```
```suggestion
      maintainers = with maintainers; [ felixsinger ];
```

felixsinger commented:
@NickHu Done

NickHu reviewed (changes requested):
I guess the final thing to do is change the commit and PR title to something like

`mfc9332cdw: init at 1.1.3-0`

There is also no corresponding entry in `pkgs/top-level/all-packages.nix`, so how exactly are you building and using this? There are a bunch of Brother drivers scattered in `pkgs/misc/cups/drivers`, but you're right that they should probably sit in their own subdirectory. There is certainly a lot of cruft in this section of nixpkgs and it could do with a cleanup...

felixsinger commented:
@NickHu Done. I adjusted the commit message and added the packages to all-packages.nix.

Until now I wasn't aware of all-packages.nix. To build the packages I used callPackage with nix-build and I added a line to my configuration.nix to test it.

flokli commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25948,6 +25948,9 @@ in
 
   hll2390dw-cups = callPackage ../misc/cups/drivers/hll2390dw-cups { };
 
+  mfc9332cdwcupswrapper = (callPackage ../misc/cups/drivers/brother/mfc9332cdw { }).mfc9332cdwcupswrapper;
```
```suggestion
  mfc9332cdw-cupswrapper = (callPackage ../misc/cups/drivers/brother/mfc9332cdw { }).cupswrapper;
```

to match the attribute name below

flokli commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+  mfc9332cdwcupswrapper = stdenv.mkDerivation rec {
+    version = "1.1.4-0";
+    name = "${model}cupswrapper-${version}";
+    dlfid = "dlf101642";
+    relpath_wrapper = "${relpath_printer}/cupswrapper";
```
These can probably also be moved to a let expr, like with the drv above.

flokli commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+      mkdir -p $out/var/tmp
+
+      # copy files to ${relpath_wrapper}
+      cp brcupsconfig/brcupsconfpt1 $out/${relpath_wrapper}
+      cp cupswrapper/cupswrapper${model} $out/${relpath_wrapper}
+      cp PPD/brother_${model}_printer_en.ppd $out/${relpath_wrapper}
+
+      # configure permissions
+      chmod 755 $out/${relpath_wrapper}/brcupsconfpt1
+      chmod 755 $out/${relpath_wrapper}/cupswrapper${model}
+      chmod 644 $out/${relpath_wrapper}/brother_${model}_printer_en.ppd
```
you can use `install -m …` to copy these over while specifying the chmod.
There's also `-D`, which you might be able to use to create the directories in first place.

flokli commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+  mfc9332cdwlpr = pkgsi686Linux.stdenv.mkDerivation rec {
+    version = "1.1.3-0";
+    name = "${model}drv-${version}";
+    dlfid = "dlf101620";
+    relpath_lpd = "${relpath_printer}/lpd";
```
move these into the above let expr, so it's more readable what other variables are available inside the `mkDerivation`.

flokli commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+
+    nativeBuildInputs = [ dpkg makeWrapper ];
+
+    unpackPhase = "dpkg-deb -x $src $out";
```
use `unpackCmd` instead of overriding the whole phase.

flokli commented on pkgs/misc/cups/drivers/brother/mfc9332cdw/default.nix:
```diff
@@ -0,0 +1,150 @@
+    };
+  };
+
+  mfc9332cdwcupswrapper = stdenv.mkDerivation rec {
```
```suggestion
  cupswrapper = stdenv.mkDerivation rec {
```

mweinelt commented:
Hey @felixsinger, it's been a while. Are you planning to address the recommendations made by @flokli?

SuperSandro2000 commented:
Closing due to inactivity from author.
